src/fides/api/util/tcf/tcf_experience_contents.py

```diff
@@ -171,21 +175,28 @@ def get_matching_privacy_declarations(db: Session) -> Query:
             System.privacy_policy.label("system_privacy_policy"),
             System.vendor_id,
             PrivacyDeclaration.data_use,
-            PrivacyDeclaration.legal_basis_for_processing,
+            PrivacyDeclaration.overridden_legal_basis_for_processing.label(  # pylint: disable=no-member
+                "legal_basis_for_processing"
+            ),
             PrivacyDeclaration.features,
             PrivacyDeclaration.retention_period,
+            PrivacyDeclaration.flexible_legal_basis_for_processing,
+            PrivacyDeclaration.purpose,
+            PrivacyDeclaration.legal_basis_for_processing.label(
+                "original_legal_basis_for_processing"
+            ),
         )
         .outerjoin(PrivacyDeclaration, System.id == PrivacyDeclaration.system_id)
         .filter(
             or_(
                 and_(
                     GVL_DATA_USE_FILTER,
-                    PrivacyDeclaration.legal_basis_for_processing
+                    PrivacyDeclaration.overridden_legal_basis_for_processing
                     == LegalBasisForProcessingEnum.CONSENT,
                 ),
                 and_(
                     GVL_DATA_USE_FILTER,
-                    PrivacyDeclaration.legal_basis_for_processing
+                    PrivacyDeclaration.overridden_legal_basis_for_processing
```

Comment:
Changes were simple here once the `overridden_legal_basis_for_processing` class-level hybrid property was working.

Reply:
i see - so the fact that this is `None` in the case of an exclusion override naturally filters those out.

this does seem  _efficient_ (in terms of being a very targeted change), i just get a bit concerned about that behavior being opaque moving forward. relates to my comment above, still thinking about this 🤔